hack/verify-shellcheck.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly VERSION="v0.9.0"
+
+echo "Installing shellcheck..."
+#this install.sh file is a script i created to simplify the instalation of shellcheck
+#it is a fork from this script https://github.com/golangci/golangci-lint/blob/master/install.sh
+#which is used on the verify-golint.sh script from this repo
+curl -sfL https://raw.githubusercontent.com/gabrielggg/shellcheck/master/install.sh| sh -s "$VERSION"
```

Comment:
So I would like to avoid adding in an extra dep on your personal repository here, can we copy over just the relevant bits from `install.sh`? 

More specifically I don't think we need to support windows, we can fix the version of shellcheck being used, and we can assume that curl is there. 

It seems like we _should_ be able to just download the statically compiled binary for linux from https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz and then run it within a script! 

With the main goal being, let's keep it as simple as possible :)